### DISCUSSION TITLE
Assign function works inline but not in options:action

### DISCUSSION THREAD

**Discussion by tomByrer** [statelyai/xstate#1500] [General] [answered]
This works:
```js
SELECT: {
  target: 'filled',
  actions: assign((context, event) => ({
    selected: event.selected
  }))
},
```

But is does not work:
```js
actions: {
  selecting: assign((context, event) => ({
    selected: event.selected
  }))
},
```
I'd like to DRY & be able to update the context:Selected from both states.  Worked on this for over an hour & I'm stuck :/
Any help or other examples would be nice.  TIA

**Comment by Andarist** [comment, marked as answer]
It works - you have just made a mistake. `actions` is a part of the second argument of the `Machine`/`createMachine` and not part of the first one. So your machine should look like this:
```js
const radioGroup = Machine(
  {
    id: "radioGroup",
    initial: "emptied",
    context: {
      selected: "",
    },
    states: {
      emptied: {
        on: {
          SELECT: {
            target: "filled",
            actions: ["selecting"],
          },
        },
      },
      filled: {
        on: {
          SELECT: {
            actions: ["selecting"],
          },
        },
      },
    },
  },
  {
    actions: {
      selecting: assign((context, event) => ({
        selected: event.selected,
      })),
    },
  }
);
```

**Reply by davidkpiano** [reply]
Note to self: dev warnings can probably help with this.

**Reply by tomByrer** [reply]
Works, cheers @Andarist !

@davidkpiano I might have gotten lost in the bracket depths.  Composing or [bracket colorizing](https://marketplace.visualstudio.com/items?itemName=CoenraadS.bracket-pair-colorizer-2) could be cheaper improvements.
I also was looking for other simple examples to help out, but as this & Spectrum forums grow, there will be more.
